inference-engine/thirdparty/clDNN/src/batch_to_space.cpp

```diff
@@ -34,88 +34,56 @@ layout batch_to_space_inst::calc_output_layout(batch_to_space_node const& node)
 
     auto input_layout = node.input(0).get_output_layout();
     auto input_format = input_layout.format;
+    auto output_format = input_format;
 
-    const size_t dims_num = format::dimension(input_format);
-
-    // Getting data from constant inputs. There are 3 args: block_shape, crops_begin, crops_end
-    std::vector<std::vector<int32_t>> args;
-    for (size_t i = 1; i < node.get_dependencies().size(); ++i) {
-        auto& input = node.get_dependency(i).as<data>();
-        auto& mem = input.get_attached_memory();
-        std::vector<int32_t> sizes;
-        if (input.get_output_layout().data_type == cldnn::data_types::i64) {
-            int64_t* data = static_cast<int64_t*>(mem.lock());
-            std::vector<int64_t> sizes_i64 = std::vector<int64_t>(data, data + input.get_output_layout().count());
-            sizes.resize(sizes_i64.size());
-            for (size_t j = 0; j < sizes.size(); j++)
-                sizes[j] = static_cast<int32_t>(sizes_i64[j]);
-        } else {
-            int32_t* data = static_cast<int32_t*>(mem.lock());
-            sizes = std::vector<int32_t>(data, data + input.get_output_layout().count());
-        }
-        args.push_back(sizes);
-        mem.unlock();
+    if (output_format == format::b_fs_yx_fsv16) {
+        output_format = format::bfyx;
     }
+    auto output_type = desc->output_data_type ? *desc->output_data_type : input_layout.data_type;
+
+    if (node.has_fused_primitives())
+        output_type = node.get_fused_output_layout().data_type;
+
+    const size_t dims_num = format::dimension(input_format);
 
-    const auto& block_shape = args[0];
-    const auto& crops_begin = args[1];
-    const auto& crops_end = args[2];
+    const auto& block_shape = desc->block_shape;
+    const auto& crops_begin = desc->crops_begin;
+    const auto& crops_end = desc->crops_end;
 
-    if (block_shape[0] != 1)
+    if (block_shape.batch[0] != 1)
         CLDNN_ERROR_MESSAGE(node.id(),
             "block_shape[0] is expected to be 1. Actual block_shape[0] is " +
-            std::to_string(block_shape[0]));
+            std::to_string(block_shape.batch[0]));
 
-    if (crops_begin[0] != 0)
+    if (crops_begin.batch[0] != 0)
         CLDNN_ERROR_MESSAGE(node.id(),
             "crops_begin[0] is expected to be 0. Actual crops_begin[0] is " +
-            std::to_string(crops_begin[0]));
+            std::to_string(crops_begin.batch[0]));
 
-    if (crops_end[0] != 0)
+    if (crops_end.batch[0] != 0)
         CLDNN_ERROR_MESSAGE(node.id(),
             "crops_end[0] is expected to be 0. Actual crops_end[0] is " +
-            std::to_string(crops_end[0]));
+            std::to_string(crops_end.batch[0]));
 
-    size_t block_sizes_multiplied = 1;
-    for (const auto& el: block_shape)
-        block_sizes_multiplied *= el;
+    size_t block_sizes_multiplied = block_shape.feature[0];
+    for (size_t i = 0; i < dims_num - 2; ++i)
```

Comment:
If I understand correctly, "- 2" means excluding Batch and Feature channels to get spatial dims number. In this case it is better to use `spatial_num(format)` function